### ISSUE TITLE
BINDetect Key Error exception

### ISSUE THREAD

**Issue by kdemuren** [loosolab/TOBIAS#10]
I was able to run ATACorrect and ScoreBigwig fine on my samples, but when I got to BINDetect, using this command:
`TOBIAS BINDetect --motifs /net/bmc-pub9/data/boyerlab/users/kdemuren/output-atac/test_motifs.txt --signals /net/bmc-pub9/data/boyerlab/users/kdemuren/output-atac/tobias_out/ESC_ATAC.footprints.bw --genome /net/bmc-pub9/data/boyerlab/users/kdemuren/mm9.fa --peaks /net/bmc-pub9/data/boyerlab/users/kdemuren/output-atac/ATAC_allPeaks.bed --cond-names ESC --outdir tobias_out/CM_motifs --cores 8`

I get this error:
```
Problem in main logger process:
Traceback (most recent call last):
  File "/home/kdemuren/miniconda3/envs/encode-atac-seq-pipeline/lib/python3.7/site-packages/tobias/utils/logger.py", line 151, in main_logger_process
    record = self.queue.get()
  File "<string>", line 2, in get
  File "/home/kdemuren/miniconda3/envs/encode-atac-seq-pipeline/lib/python3.7/multiprocessing/managers.py", line 796, in _callmethod
    kind, result = conn.recv()
  File "/home/kdemuren/miniconda3/envs/encode-atac-seq-pipeline/lib/python3.7/multiprocessing/connection.py", line 250, in recv
    buf = self._recv_bytes()
  File "/home/kdemuren/miniconda3/envs/encode-atac-seq-pipeline/lib/python3.7/multiprocessing/connection.py", line 407, in _recv_bytes
    buf = self._recv(4)
  File "/home/kdemuren/miniconda3/envs/encode-atac-seq-pipeline/lib/python3.7/multiprocessing/connection.py", line 383, in _recv
    raise EOFError
EOFError
multiprocessing.pool.RemoteTraceback: 
"""
Traceback (most recent call last):
  File "/home/kdemuren/miniconda3/envs/encode-atac-seq-pipeline/lib/python3.7/multiprocessing/pool.py", line 121, in worker
    result = (True, func(*args, **kwds))
  File "/home/kdemuren/miniconda3/envs/encode-atac-seq-pipeline/lib/python3.7/site-packages/tobias/tools/bindetect_functions.py", line 236, in scan_and_score
    bigwig_score = footprints[bigwig][pos]
KeyError: 'ESC'
"""

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/kdemuren/miniconda3/envs/encode-atac-seq-pipeline/bin/TOBIAS", line 11, in <module>
    load_entry_point('tobias==0.11.0', 'console_scripts', 'TOBIAS')()
  File "/home/kdemuren/miniconda3/envs/encode-atac-seq-pipeline/lib/python3.7/site-packages/tobias/TOBIAS.py", line 154, in main
    args.func(args)             
  File "/home/kdemuren/miniconda3/envs/encode-atac-seq-pipeline/lib/python3.7/site-packages/tobias/tools/bindetect.py", line 305, in run_bindetect
    results = [task.get() for task in task_list]
  File "/home/kdemuren/miniconda3/envs/encode-atac-seq-pipeline/lib/python3.7/site-packages/tobias/tools/bindetect.py", line 305, in <listcomp>
    results = [task.get() for task in task_list]
  File "/home/kdemuren/miniconda3/envs/encode-atac-seq-pipeline/lib/python3.7/multiprocessing/pool.py", line 657, in get
    raise self._value
KeyError: 'ESC'
```
Is there something I'm doing wrong here? Thanks!

**Comment by msbentsen** [comment]
It seems there was a problem reading scores from the .bigwig, since the 'footprints' dict never received the scores from condition 'ESC'. There could be a couple of reasons for that. Can you confirm that the chromosomes in --peaks and --signals fit together? Just to make sure that we are not trying to fetch scores for "chrX" if "chrX" does not exist in the bigwig.  

In any case, the error should be caught earlier - that is on me, so sorry about that! I am working on a patch which should give a better error message.

**Comment by kdemuren** [comment]
Thanks for the quick response! The footprints file is just the output from the FootprintScores command:
```
TOBIAS FootprintScores --signal /net/bmc-pub9/data/boyerlab/users/kdemuren/output-atac/tobias_out/ESC_ATAC.sorted_corrected.bw --regions /net/bmc-pub9/data/boyerlab/users/kdemuren/output-atac/ATAC_allPeaks.bed --output /net/bmc-pub9/data/boyerlab/users/kdemuren/output-atac/tobias_out/ESC_ATAC.footprints.bw --cores 8
```
and it's the same peaks .bed file I've been using throughout, so I don't know how a mismatch could have been introduced.

**Comment by msbentsen** [comment]
Can you check your version of the python package `pyBigWig`? You can use ` pip freeze | grep pyBigWig`. I just noticed that BINDetect uses a functionality only included in pybigwig>=0.3.

If upgrading pybigwig is not the issue, could you try to run the command with the TOBIAS version on the dev branch? This version raises the exception earlier (and stops the logger from dying so loudly!), so it should give a better idea of the actual problem.
 
Thanks!

**Comment by kdemuren** [comment]
I've got version 0.3.13 installed. Is there a way I can get the dev branch to install easily with conda? That's how I installed TOBIAS originally.

**Comment by msbentsen** [comment]
I can't make releases to conda for dev versions, but you can use the command:
`pip install git+git://github.com/loosolab/TOBIAS@dev`
This will install TOBIAS directly into your current conda environment. You can use `TOBIAS --version` to confirm that you fetched `0.11.1-dev`. Once the bug is solved I will of course make an official release on conda. Hope this works for you!

**Comment by msbentsen** [comment]
I believe that this issue is fixed by the 0.11.1 release, but please reopen if this should not be the case! ([Link to CHANGES for 0.11.1](https://github.com/loosolab/TOBIAS/commit/d04325687b9936c9f48b64d54d64706d279a6a18#diff-e4eb329834da3d36278b1b7d943b3bc9))
